File: tests/fixtures/corpus/pii-in-logs/positive.py
# Corpus fixture for pii-in-logs-python: PII reaching a logger via string
# concatenation, f-string substitution, a multi-argument call, and direct
# member access. Every log line below must produce a finding.
import logging

logger = logging.getLogger(__name__)


def audit(user, otp):
    logger.info("verification code sent to " + user.phone)
    logger.warning(f"otp {otp} issued for {user.email}")
    logger.error("failed login attempt", user.password)
    log.debug(user.ssn)

File: tests/fixtures/corpus/pii-in-logs/positive.ts
// Corpus fixture for pii-in-logs-javascript: PII reaching a logger via string
// concatenation, template-literal substitution, a multi-argument call, and
// direct member access. Every log line below must produce a finding.

interface Account {
  email: string;
  phone: string;
  password: string;
}

export function audit(account: Account, otp: string): void {
  // String concatenation
  console.log("verification code sent to " + account.phone);
  // Template-literal substitution
  logger.info(`otp ${otp} issued for ${account.email}`);
  // Multi-argument call
  logger.warn("failed login attempt", account.password);
  // Direct member access (the original rule already caught this)
  log.debug(account.email);
}
